### ISSUE TITLE
SaveChanges fails for model with owned type iff Update was called and using Sqlite

### ISSUE THREAD

**Issue by AsgerPetersen** [dotnet/efcore#31014]
This seems like surprising behaviour:
```C#
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;

// App
var context = new ReproduceContext();

context.Database.EnsureCreated();

var p = new Person()
{
    Name = "Huey",
    Active = new Period() { From = DateTime.Now, To = DateTime.MaxValue }
};
context.Person.Add(p);
context.SaveChanges(); // Works

p.Name = "Dewey";
context.SaveChanges(); // Works

p.Name = "Louie";
context.Update(p);
context.SaveChanges(); // Works with SQLServer but throws with Sqlite
```
Models, context etc for complete reproducable:

```C#
// Models
public class Period
{
    public DateTime From { get; set; }
    public DateTime To { get; set; }
}

public class Person
{
    public Guid Id { get; protected set; }
    public string Name { get; set; } = null!;
    public Period Active { get; set; } = null!;
}

// Configuration
public class PersonConfiguration : IEntityTypeConfiguration<Person>
{
    // Configures Person as a temporal table
    public void Configure(EntityTypeBuilder<Person> builder)
    {
        // Make Person table temporal
        builder.ToTable(typeof(Person).Name, tableBuilder =>
            {
                tableBuilder.IsTemporal();
                tableBuilder.Property<DateTime>("PeriodStart").HasColumnName("PeriodStart");
                tableBuilder.Property<DateTime>("PeriodEnd").HasColumnName("PeriodEnd");
            }
        );

        // Owned types need special care. Tracked in https://github.com/dotnet/efcore/issues/29303
        builder.OwnsOne(e => e.Active, ownedBuilder =>
        {
            ownedBuilder.ToTable(typeof(Person).Name,
            tableBuilder =>
            {
                tableBuilder.IsTemporal();
                tableBuilder.Property<DateTime>("PeriodStart").HasColumnName("PeriodStart");
                tableBuilder.Property<DateTime>("PeriodEnd").HasColumnName("PeriodEnd");
            });
        }
        );
    }
}

// Context
public class ReproduceContext : DbContext
{
    public DbSet<Person> Person { get; set; }

    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        modelBuilder.ApplyConfiguration(new PersonConfiguration());

        base.OnModelCreating(modelBuilder);
    }

    protected override void OnConfiguring(DbContextOptionsBuilder options)
        => options.UseSqlite($"Data Source=./reproduce.db");
    //options.UseSqlServer("Server=localhost,1433; Database=reproduce;User=sa;Password=xxx;TrustServerCertificate=True");
}
```
If the owned type (`Period`) is removed from `Person` then the above works with both SQLServer and Sqlite.

### Exception thrown
When using Sqlite the above throws this exception

```
Exception has occurred: CLR/System.InvalidOperationException
An unhandled exception of type 'System.InvalidOperationException' occurred in Microsoft.EntityFrameworkCore.dll: 'The property 'Period.PeriodEnd' cannot be assigned a value generated by the database. Store-generated values can only be assigned to properties configured to use store-generated values.'
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetStoreGeneratedValue(IProperty property, Object value, Boolean setModified)
   at Microsoft.EntityFrameworkCore.Update.ModificationCommand.ColumnValuePropagator.TryPropagate(IColumnMappingBase mapping, IUpdateEntry entry)
   at Microsoft.EntityFrameworkCore.Update.ModificationCommand.<GenerateColumnModifications>g__HandleColumnModification|42_5(IColumnMappingBase columnMapping, <>c__DisplayClass42_0& , <>c__DisplayClass42_1& , <>c__DisplayClass42_2& )
   at Microsoft.EntityFrameworkCore.Update.ModificationCommand.GenerateColumnModifications()
   at Microsoft.EntityFrameworkCore.Update.ModificationCommand.<>c.<get_ColumnModifications>b__34_0(ModificationCommand command)
   at Microsoft.EntityFrameworkCore.Internal.NonCapturingLazyInitializer.EnsureInitialized[TParam,TValue](TValue& target, TParam param, Func`2 valueFactory)
   at Microsoft.EntityFrameworkCore.Update.ModificationCommand.get_ColumnModifications()
   at Microsoft.EntityFrameworkCore.Update.Internal.CommandBatchPreparer.<CreateCommandBatches>d__11.MoveNext()
   at Microsoft.EntityFrameworkCore.Update.Internal.CommandBatchPreparer.<BatchCommands>d__9.MoveNext()
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.Execute(IEnumerable`1 commandBatches, IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Storage.RelationalDatabase.SaveChanges(IList`1 entries)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(IList`1 entriesToSave)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(StateManager stateManager, Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.<>c.<SaveChanges>b__107_0(DbContext _, ValueTuple`2 t)
   at Microsoft.EntityFrameworkCore.Storage.NonRetryingExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges()
   at Program.<Main>$(String[] args) in /Users/asger/Code/Aika/Aika-DevHelpers/EfCorePeriodeRepro/Program.cs:line 22
```

### Provider and version information

EF Core version: 7.0.5
Database provider: Microsoft.EntityFrameworkCore.Sqlite
Target framework: .NET 7.0
Operating system: Darwin x64 22.4.0
IDE: VSCode 1.78.2

**Comment by ajcvickers** [comment]
Note for triage: issue here appears to be a consequence of two types mapped to the same table, where columns in that table are mapped to different properties in each type. Same behavior happens on SQL Server and SQLite. (The posted code works on SQL Server because the two properties are configured as temporal and hence handled differently by the update pipeline. The exception is the same as on SQLite when temporal tables are not used.)

**Comment by AndriySvyryd** [comment]
Fixed in 11bced2a45ebd11c5d2cfe1f58357248256415d4
